src/sys/system.h

```diff
@@ -148,29 +162,41 @@ class System
     static uint32_t GetPClk2Freq();
 
     /**
-     ** Returns a const reference to the Systems Configuration struct
+     ** Returns a const reference to the Systems Configuration struct.
      */
     const Config& GetConfig() const { return cfg_; }
 
-    /** Returns an enum representing the current (primary) memory space used
+    /** Returns an enum representing the current (primary) memory space used 
      *  for executing the program.
      */
-    static ProgramMemory GetProgramMemory();
+    static MemoryRegion GetProgramMemoryRegion();
+
+    /** Returns an enum representing the the memory region 
+     *  that the given address belongs to.
+     *  \param address The address to be checked
+     */
+    static MemoryRegion GetMemoryRegion(uint32_t address);
+
+    /** This offset from the base QSPI address is reserved
+     *  for libDaisy functionality.
+     */
+    static constexpr uint32_t kQspiOffset = 0x40000U;
```

Comment:
This almost makes it seem like users _shouldn't_ use the first 256kB, but it's actually that the first 0x40000 are left for user data.